File: han_dog_message/dart/example/cms_client.dart
import 'package:grpc/grpc.dart';
import 'package:han_dog_message/han_dog_message.dart' as msg;

void main() async {
  final channel = ClientChannel(
    'localhost',
    port: 13145,
    options: const ChannelOptions(credentials: ChannelCredentials.insecure()),
  );
  final stub = msg.CmsClient(channel);
  final response = await stub.enable(msg.Empty());
  print('Enable response: $response');
  // await for (var status in stub.listenStrategy(msg.Empty())) {
  //   print('Strategy: ${status.data}, Timestamp: ${status.timestamp.toDart()}');
  // }
  await channel.shutdown();
}

File: han_dog_message/python/example/cms_client.py
import asyncio
from grpc import aio
import han_dog_message as msg


async def main():
    async with aio.insecure_channel("127.0.0.1:13145") as channel:
        stub = msg.CmsStub(channel)
        response = await stub.Enable(msg.Empty())
        print(f"Enable response received {response}")
        async for r in stub.ListenStrategy(msg.Empty()):
            print(f"Strategy: {r}")


if __name__ == "__main__":
    asyncio.run(main())
